### ISSUE TITLE
Crash when starting a webm recording on webgl

### ISSUE THREAD

**Issue by MonsieurD** [natmlx/natcorder#18]
Hi,

I have a javascript crash on my webgl build when i try to record a video. Here are the logs from the navigator console :

```
NatCorder: WEBMRecorder prepared video encoder at resolution 720x1280@30.000000Hz with average bitrate 10000000

Invoking error handler due to
TypeError: handle[name] is not a function
methodCaller_emscripten$$val_$$@https://xxxxxxxxx/NextDancerPP_Git2324_Release_IL2CPP/NextDancerPP_IL2CPP/Build/NextDancerPP_IL2CPP.framework.js line 3 > Function:4:26
__emval_call_method@https://xxxxxxxxx/NextDancerPP_Git2324_Release_IL2CPP/NextDancerPP_IL2CPP/Build/NextDancerPP_IL2CPP.framework.js:3:189560
@https://xxxxxxxxx/NextDancerPP_Git2324_Release_IL2CPP/NextDancerPP_IL2CPP/Build/NextDancerPP_IL2CPP.framework.js line 3 > WebAssembly.instantiate:wasm-function[52905]:0x10481b1
@https://xxxxxxxxx/NextDancerPP_Git2324_Release_IL2CPP/NextDancerPP_IL2CPP/Build/NextDancerPP_IL2CPP.framework.js line 3 > WebAssembly.instantiate:wasm-function[52878]:0x10463d2

```

Here is my code :

```
                m_Clock = new RealtimeClock();

#if UNITY_WEBGL && !UNITY_EDITOR
                m_Recorder = new WEBMRecorder(c_RecordWidth, c_RecordHeight, c_RecordFrameRate, AudioSettings.outputSampleRate, (int)AudioSettings.speakerMode);
#else
                m_Recorder = new MP4Recorder(c_RecordWidth, c_RecordHeight, c_RecordFrameRate, AudioSettings.outputSampleRate, (int)AudioSettings.speakerMode);
#endif

                m_CameraInput = new CameraInput(m_Recorder, m_Clock, m_Camera);
                m_AudioInput = new AudioInput(m_Recorder, m_Clock, AudioManager.Instance.Listener);

                m_Recording = true;
```

c_RecordWidth = 720
c_RecordHeight = 1280
c_RecordFrameRate = 30

audio listener and camera are not null.

Am I doing something wrong ?

**Comment by olokobayusuf** [comment]
Hey what version of Unity are you recording on? Also, Unity doesn't support `OnAudioFilterRead` on WebGL, so you can't use an `AudioInput` to record engine audio in WebGL. [See the docs](https://docs.natml.ai/natcorder/api/recorder-inputs/audioinput).

**Comment by MonsieurD** [comment]
It's Unity 2021.3.13f1.

Can I use the CommitSamples method to record audio ?

**Comment by olokobayusuf** [comment]
Hm okay I'll try to repro. Yes, you can use `CommitSamples` if you already have audio data in hand (this is how we record the mic with NatDevice).

**Comment by MonsieurD** [comment]
Can I get the audio data from the unity audio listener ?

**Comment by olokobayusuf** [comment]
> Can I get the audio data from the unity audio listener ?

This is the purpose of `OnAudioFilterRead`, and as I mentioned earlier, Unity doesn't support this on WebGL.

I was able to reproduce this error when creating a `WEBMRecorder` with audio recording. I'll keep this issue open till we resolve the error.

**Comment by olokobayusuf** [comment]
[Fixed in NatCorder 1.9.4](https://github.com/natmlx/natcorder/releases/tag/1.9.4).
